### PR TITLE
Implement `DeepHashAccessor` for accessing seam resources' nested hashes

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
`DeepHashAccessor` enhances our SDK by providing intuitive dot notation access to nested data structures, automatically handling arbitrary depths of hashes and arrays.

- **Add DeepHashAccessor and use it for accessing nested hashes in BaseResource**
- **Add tests**

### CODE REVIEW

File: lib/seam/base_resource.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require_relative "utils/deep_hash_accessor"
```
Comment (razor-x) [inline lib/seam/base_resource.rb:3]:
```suggestion
require_relative "./deep_hash_accessor"
```

File: spec/utils/deep_hash_accessor_spec.rb
Comment (razor-x) [inline spec/utils/deep_hash_accessor_spec.rb:1]: For unit tests of lib tools like this, we should be able to put the test next to the file under `lib/seam`

**Review by razor-x — APPROVED** [review]

File: lib/seam/base_resource.rb (hunk 1 above)
Comment (andrii-balitskyi) [reply lib/seam/base_resource.rb]: 2cec68f6573cb4902aa1069f5e14142cbb75c1cb

File: spec/utils/deep_hash_accessor_spec.rb
Comment (andrii-balitskyi) [reply spec/utils/deep_hash_accessor_spec.rb:1]: bfa9467a94396550490d0365dd876a38db4e9071

File: lib/seam/base_resource.rb (hunk 1 above)
Comment (andrii-balitskyi) [reply lib/seam/base_resource.rb]: Should we move `lib/seam/utils/action_attempt_utils.rb` out of utils as well?

**Pull request merged**